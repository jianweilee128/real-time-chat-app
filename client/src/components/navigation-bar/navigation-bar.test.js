import React from "react";
import { shallow } from "enzyme";
import NavigationBar from "./navigation-bar.component";

describe("NavigationBar component", () => {
  let wrapper;
  let mockSetToggleOptionsPopup;
  let mocktoggleUpdateProfile;

  beforeEach(() => {
    mockSetToggleOptionsPopup = jest.fn();
    mocktoggleUpdateProfile = jest.fn();

    const mockProps = {
      user: { name: "JohnDoe", id: "123" },
      currentRoom: [("mockRoom", "123")],
      roomList: [
        {
          name: "mockRoom",
          _id: "123",
        },
      ],
      setToggleOptionsPopup: mockSetToggleOptionsPopup,
      toggleUpdateProfile: mocktoggleUpdateProfile,
    };

    wrapper = shallow(<NavigationBar.WrappedComponent {...mockProps} />);
  });

  it("should render NavigationBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setToggleOptionsPopup function when clicked", () => {
    wrapper.find(".add-icon").simulate("click");
    expect(mockSetToggleOptionsPopup).toHaveBeenCalled();
  });
  it("should call toggleUpdateProfile function when clicked", () => {
    wrapper.find(".edit-icon").simulate("click");
    expect(mocktoggleUpdateProfile).toHaveBeenCalled();
  });
});
