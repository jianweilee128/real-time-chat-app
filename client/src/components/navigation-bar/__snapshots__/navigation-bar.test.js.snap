// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar component should render NavigationBar component 1`] = `ShallowWrapper {}`;
